### ISSUE TITLE
Survey info from client not saving

### ISSUE THREAD

**Issue by aqw** [psychoinformatics-de/dmc#31]
FML

**Comment by yarikoptic** [comment]
Do you mean that those which files it out yesterday were not saved?

**Comment by aqw** [comment]
The records were saved, with none of the content. :-(

We missed out on half a dozen (maybe a few more).

---Alex

**Comment by yarikoptic** [comment]
That truly sucks since some of those were super informative (a right tail on the distribution) and I would hate to ask those folks to resubmit... heh

**Comment by yarikoptic** [comment]
And by content do you mean just custom answers or all choice selections too?

**Comment by aqw** [comment]
@yarikoptic Yeah, I'm not happy about it either. It was a last minute change to support Safari and IE that IMO should have caused a JS parsing error but didn't. The peculiarities of the way browsers implements/fail on the spec. Did I mention I'm not happy?

When I say no content, I mean just that. /Nothing/ from the client was ever sent to the server.

---Alex

**Comment by yarikoptic** [comment]
On a positive side good that it was only the first wave ;-) but now it is all kosher as far as you can see, and I could ask few to resubmit?
Btw in handout we should have included URL in addition to qr code ;-) write it down manually for now

**Comment by aqw** [comment]
@yarikoptic I'm not feeling well at all, so I'm doing my best curb a sarcastic response.

Yes, to the best of my knowledge, everything is working now. I'm not sure why you would think that I would find a problem, fix it, test it, inform you, feel shitty about it, continue testing, find something, /not/ fix it, /not/ tell you, and then have you continue running the survey. I'm not seeing your line of thought.

As for the URL, if I don't feel like vomiting tomorrow, I'll haul my ass to work again and make a version of the flyers with the URL on it.

---Alex

**Comment by yarikoptic** [comment]
Sorry to hear that you aren't feeling well :-/  Well -- here the entire country seems to be a bit sick ATM ;)

As for "why" -- it was not to be sarcastic and definitely was not intended to annoy you.  I was re-confirming just because there is no automated testing ATM and the only assurance for me  could be your words.  But we are all humans and I am not sure how wide could your manual testing be.  May be you have just fixed it and tested only for this specific issue on just few browsers/platforms but did not test other functionality and have missed some other side-effects?  or you could have done really extensive manual testing going through a variety of the use-cases, replicating your previous testing efforts... I don't know, and that is why was seeking re-confirmation ;)

don't bother with new fliers for this occasion -- I will just write url down on those which I have printed.  I brought it up only so that next time we (I have missed it too, you know) do not forget a url on the fliers -- many folks still do not have any qrcode reader installed on their phones, and only some phones (samsungs?) provide qr code reading out of the box with stock camera app.

**Comment by aqw** [comment]
Sorry for the snark. It just came across as insulting.

As for how wide was my testing of this fix: not very.

Prior to the Safari/IE fixes, I did wide testing. Post-fixes, I thought I tested everything, but I obviously missed checking the /content/ of the files being saved.

With this fix, I filled out the form, ensured no JS errors, made sure the form content is sent, and made sure it is saved server-side. I tested on Chromium and Firefox on Debian and Safari on OS X. But, the fix is old-school, bone-headed-featureless-stupid JS, so I expect it to work everywhere.

I did not do a line-by-line check to make sure each question is being saved properly. However, I did check every question-type (hidden, select, checklist, dynamic checklist).

I did not test any functionality that shouldn't be affected by this change. You're right, without automated testing, I can't test everything, everywhere, under every configuration.

Good to know about the QR codes. I don't have a smartphone, so I'm not part of that world. I see QR codes everywhere and Michael specifically asked for it. So I just assumed it was common.

---Alex
